storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -4166,4 +4184,23 @@ public void shutdown() {
     public boolean isWaiting() {
         return timer.isTimerWaiting();
     }
+
+    @Override
+    public void processWorkerMetrics(WorkerMetrics metrics) throws org.apache.thrift.TException {
+        if (this.metricsStore == null) {
```

Comment:
Can we add in some authorization calls before we do anything with executing the command?

```
checkAuthorization(null, null, "processWorkerMetrics");
```

Then we would need to update 

https://github.com/apache/storm/blob/7ecb3d73e8e909c01d39e03a7a7ed45a2fb81859/storm-client/src/jvm/org/apache/storm/security/auth/authorizer/SimpleACLAuthorizer.java#L52

to have processWorkerMetrics in the list.